### ISSUE TITLE
FULL OUTER JOIN doesn't work properly

### ISSUE THREAD

**Issue by CheyenneForbes** [crate/crate#8815]
**CrateDB version**:
4.0.0 -SNAPSHOT-4ce5528

**CrateDB issue**:
here's the schema

http://sqlfiddle.com/#!17/b1e3e/1

in crate it returns:

barid | fooid | barid | isapproved
-- | -- | -- | --
2 | 2 | 2 | 1
3 | 4 | 3 | 1
3 | 5 | 3 | 1

instead of (ignore the column order):

fooid | barid | barid | isapproved
-- | -- | -- | --
2 | 2 | 2 | 1
4 | 3 | 3 | 1
5 | 3 | 3 | 1
(null) | (null) | 33 | 1
(null) | (null) | 34 | 1
(null) | (null) | 36 | 1

**Comment by seut** [comment]
@CheyenneForbes Thanks for reporting, we'll have a closer look asap.

Btw. with latest nightly, the output is a bit different but still not correct ;) 

```
+-------+-------+-------+------------+
| fooid | barid | barid | isapproved |
+-------+-------+-------+------------+
|     6 |  NULL |  NULL |       NULL |
|     3 |  NULL |  NULL |       NULL |
|     1 |     1 |  NULL |       NULL |
|     2 |     2 |     2 |          1 |
|     4 |     3 |     3 |          1 |
|     5 |     3 |     3 |          1 |
|  NULL |  NULL |    33 |          1 |
|  NULL |  NULL |    36 |          1 |
|  NULL |  NULL |    34 |          1 |
+-------+-------+-------+------------+
```

**Comment by CheyenneForbes** [comment]
@seut  did you find the issue?

**Comment by seut** [comment]
@CheyenneForbes We've fixed this issue on MASTER by https://github.com/crate/crate/pull/8826 and backported it to 3.3. as well. The latest nightly does already contain the fix, while on 3.3 it will be published with the next hotfix release.
Thanks again for reporting.
